src/lib/bootloader/stage1.rb

```diff
@@ -114,11 +114,17 @@ def boot_disk_names
     end
 
     def boot_partition?
-      include_real_devs?(boot_partition_names)
+      names = boot_partition_names
+      return false if names.empty?
+
+      include_real_devs?(names)
     end
 
     def mbr?
-      include_real_devs?(boot_disk_names)
+      names = boot_disk_names
+      return false if names.empty?
+
+      include_real_devs?(names)
```

Comment:
Why it is not possible to make `include_real_devs` return false if `names` is empty?

Reply:
Well, it is possible, but inconsistent with ruby include which I would like to avoid:

```
irb(main):001:0> "abc".include?("")
=> true
irb(main):002:0> "abc".include?("a")
=> true
```

Reply:
Well, I was more thinking about an array:

```ruby
irb(main):001:0> ["a", "b", "c"].include?("")
=> false
```

Reply:
this is kind of different as it is if array include empty array and not empty string. Here names are also array and not string.